src/lib.rs

```diff
@@ -463,7 +463,7 @@ impl fmt::Debug for PctStr {
 pub trait Encoder {
 	/// Decide if the given character must be encoded.
 	///
-	/// Note that the character `%` is always encoded even if this method returns `false` on it.
+	/// Note that the character `%` MUST always be encoded.
```

Comment:
I think I have not been clear enough last round: In the current API, this method does not **have** to return `true` on `%`.  `PctString::encode` **will** encode the `%` char whatever the result of this function is, so that the correctness of the encoding is preserved independently of the implementation. What I was excepting is not a change in the doc, but a change in your proposed implementation of `PctString::encode`, because it currently removes this fail safe.